liquidator/liquidator.js

```diff
@@ -91,25 +91,98 @@ class Liquidator {
-      message: "Checking for liquidations",
-      inputPrice: priceFeed
+      message: "Checking for expired and disputed liquidations to withdraw rewards from"
     });
 
     // Update the client to get the latest information.
     await this.empClient._update();
 
-    // TODO: Just showing an example of how I want to use the client:
-    // Get expired liquidations from the client.
+    let withdrawPromises = []; // store all promises to resolve in parallel later on.
+
+    // Get all expired liquidations from the client.
     const expiredLiquidations = this.empClient.getExpiredLiquidations();
-    // TODO: 
-    // - Withdraw rewards
-    // - Check that amount of rewards was correct.
+    if (expiredLiquidations.length > 0) {
+      Logger.info({
+        at: "liquidator",
+        message: "expired liquidations detected!",
+        number: expiredLiquidations.length,
+        expiredLiquidations: expiredLiquidations
+      });
+    } else {
+      Logger.info({
+        at: "liquidator",
+        message: "No expired liquidations"
+      });
+    }
+    for (const liquidation of expiredLiquidations) {
+      Logger.info({
+        at: "liquidator",
+        message: "withdrawing rewards from expired liquidation",
+        address: liquidation.sponsor,
+        id: liquidation.id
+      });
+
+      // Confirm that liquidation has eligible rewards to be withdrawn.
+      let withdrawPromise = this.empContract.methods.withdrawLiquidation(liquidation.id, liquidation.sponsor)
+      let withdrawAmount = await withdrawPromise.call({ from: this.account })
+      if (toBN(withdrawAmount.rawValue).gt('0')) {
+        // TODO(#1062): Determine this gas price dynamically.
+        withdrawPromises.push(
+          withdrawPromise.send({ from: this.account, gas: 1500000 })
+        )
+      }
+    }
 
-    // Get disputed liquidations from the client.
+    // Get all disputed liquidations from the client.
     const disputedLiquidations = this.empClient.getDisputedLiquidations();
-    // TODO: 
-    // - Withdraw rewards
-    // - Check whether it was a successful or failed dispute and double check that the reward amount was correct
+    if (disputedLiquidations.length > 0) {
+      Logger.info({
+        at: "liquidator",
+        message: "disputed liquidations detected!",
+        number: disputedLiquidations.length,
+        disputedLiquidations: disputedLiquidations
+      });
+    } else {
+      Logger.info({
+        at: "liquidator",
+        message: "No disputed liquidations"
+      });
+    }
+    for (const liquidation of disputedLiquidations) {
+      Logger.info({
+        at: "liquidator",
+        message: "withdrawing rewards from disputed liquidation",
+        address: liquidation.sponsor,
+        id: liquidation.id
+      });
+
+      // Confirm that liquidation has eligible rewards to be withdrawn.
+      let withdrawPromise = this.empContract.methods.withdrawLiquidation(liquidation.id, liquidation.sponsor)
```

Comment:
tiny nit: consider naming this variable something else since it's not a promise. Maybe just `withdraw` or `withdrawal`...?

Reply:
I agree, would you say it is more of a "withdrawTransaction" that we can convert into a promise via `.send()`? out of curiosity

I renamed to `withdraw` for brevity.